app/controllers/fhir/r4/api_controller.rb

```diff
@@ -1044,6 +1067,10 @@ def search_patients(options)
         query = query.where('primary_telephone like ?', Phonelib.parse(search, 'US').full_e164)
       when 'email'
         query = query.where('email like ?', "%#{search}%")
+      when 'state-local-id'
```

Comment:
Unfortunately the rules of FHIR are somewhat strict about what these search parameters are supposed to be. So we can't just define `state-local-id` as a param, instead we want to use `identifier`, which is an allowed param (see [here](https://www.hl7.org/fhir/patient.html#search)). An `identifier` is searched for using the syntax `[system]|[value]`, so in this case, we will want to search based on state local ID if the param is identifier and it has the form `http://saraalert.org/SaraAlert/state-local-id|[value]`, since that is the `system` we have defined. If the `system` is something else, we can just ignore the param. If you want, you can make it so that the other systems we use for the other types of identifiers can also be searched by, but don't consider that a requirement.

Reply:
https://github.com/SaraAlert/SaraAlert/pull/1098/commits/b335e9bae20603fd40a4b1f465c206aff9f4b9ef#diff-e79e38176ee697a8541cffba23c60d9decaa69382ca5c6e875da033508e5b428R1080